core/network/src/chain.rs

```diff
@@ -78,6 +78,12 @@ pub trait FinalityProofProvider<Block: BlockT>: Send + Sync {
 	fn prove_finality(&self, for_block: Block::Hash, request: &[u8]) -> Result<Option<Vec<u8>>, Error>;
 }
 
+impl<Block: BlockT> FinalityProofProvider<Block> for () {
```

Comment:
I think the finality proof provider is optional when creating the service.

Reply:
I'll try to provide a reproducible test case, but I think what I had trouble with is that if I remove `.with_finality_proof_provider` for the service builder, the whole type resolution will fail. But maybe we should fix that instead.

Reply:
Yeah I see, it will fail to infer the generic type of the optional finality proof provider.